File: 0216-combination-sum-iii/0216-combination-sum-iii.cpp
class Solution {
public:
    vector<vector<int>> ans;
    vector<int> temp;
void recursion(  int k, int n, int i)
{
    if(k==0 && n==0)
    {     ans.push_back(temp);
            return;
    }
    
    for(int ind=i;ind<=9;ind++)
    {
        temp.push_back(ind);
        recursion(k-1,n-ind,ind+1);
        temp.pop_back();
    }
    
}
    vector<vector<int>> combinationSum3(int k, int n) {
         recursion(k,n,1);
    return ans;
    }
};

File: 0216-combination-sum-iii/0216-combination-sum-iii.java
class Solution {
    public List<List<Integer>> combinationSum3(int k, int n) {
          Set<List<Integer>> set = new HashSet<>();
        combinationHelperSum3(set, k, n, new HashSet<>());
        return new ArrayList<>(set);
    }
    public void combinationHelperSum3(Set<List<Integer>> ans, int noofelements, int remainingSum,
            Set<Integer> runningSet) {
        if (noofelements == 0) {
            if (remainingSum == 0) {
                ans.add(new ArrayList<>(runningSet));
                return;
            } else {
                return;
            }
        }
        if (remainingSum < 0) {
            return;
        }

        for (int i = 1; i <= 9; i++) {
            if(!runningSet.contains(i)){
                runningSet.add(i);
                combinationHelperSum3(ans, noofelements-1, remainingSum-i, runningSet);
                runningSet.remove(i);
            }
        }
    }
}